### ISSUE TITLE
`Exam mode`: Problem statement diff view does only show diff to last change

### ISSUE THREAD

**Issue by florian-glombik** [ls1intum/Artemis#7567]
### Is your feature request related to a problem?

If there are multiple issues in one problem statement only the latest change of the problem statement will be highlighted.

1. Mistake in problem statement found
2. Fixed problem statement
3. Notification send to students
4. Found a second mistake in the same problem statement
5. Fixed problem statement again
6. Second notification send to students
7. Students only see the second change in the problem statement

<details>
<summary>More detailed scenario</summary>

- 1 instructor
- 1 student
- exam with 1 programming exercise with code editor enabled

1. `student`: Participate in the exam
2. `instructor`: Change the problem statement
3. `student`: See that the diff is highlighted
4. `instructor`: Make another change to the problem statement
5. `student`: See that only the change about the problem statement from step 2 is displayed - but not to the initial problem statement

</details>

### Describe the solution you'd like

I think it would be the best solution to always compare to the initial problem statement _(= the one that was used when the exam started)_.

### Describe alternatives you've considered

Not supporting this edge case and presume that not more than one mistake if found within one exam exercise - or if this is the case this is stated clearly in the notification.

**Comment by florian-glombik** [comment]
This also means that we will need some kind of version-controlled problem statement _(at least the initial problems statement)_

**Comment by krusche** [comment]
Why should the diff relate to the initial problem statement? I don't think this is a bug. If at all: this would be a feature request to change the current intended functionality

**Comment by florian-glombik** [comment]
> Why should the diff relate to the initial problem statement?

I thought of a scenario with multiple changes on the same problem statement. 

>  I don't think this is a bug. If at all: this would be a feature request to change the current intended functionality

I didn't expect the feature to work this way but should have read the documentation before. I have converted the issue to a feature request now.

**Comment by krusche** [comment]
I don't think that multiple changes will take place. Ideally in exams, the problem statement is not changed at all! So this is already an exceptional situation where instructors should take a lot of care how to handle the situation even if there might be time pressure. Multiple changes would only lead to confusion and it would be difficult to display this for students in an understandable way.
